auto_update_pricing.py

```diff
@@ -0,0 +1,93 @@
+import requests
+import json
+
+def fetch_data(url):
+    """
+    Fetches data from the specified URL.
+
+    Args:
+        url (str): The URL to fetch data from.
+
+    Returns:
+        dict or None: The JSON response if successful, None otherwise.
+    """
+    try:
+        response = requests.get(url)
+        response.raise_for_status()
+        return response.json()
+    except requests.exceptions.RequestException as e:
+        print("Error fetching data from URL:", e)
+        return None
+
+def update_local_data(local_data, remote_data):
```

Comment:
I modified the function to update the models with additional fields and include new models. However, this update seems to have unintentionally changed all the pricing information to scientific notation. I don't know how to fix that.
def update_local_data(local_data, remote_data):
    """
    Updates local data with information fetched remotely.
    Args:
        local_data (dict): Local data to be updated.
        remote_data (dict): Remote data fetched from an API.
    """
    # Prepare set of existing openrouter model names for strict checking
    existing_models = {model_name for model_name in local_data if model_name.startswith("openrouter/")}

    for model in remote_data["data"]:
        model_suffix = model["id"]
        model_name = f"openrouter/{model_suffix}"

        # Calculate per token and per image costs
        input_cost_per_token = float(model["pricing"]["prompt"]) 
        output_cost_per_token = float(model["pricing"]["completion"]) 
        input_cost_per_image = float(model["pricing"]["image"]) if "image" in model["pricing"] else None


        # Preparing model data dictionary
        model_data = {
            "max_tokens": model["context_length"],
            "input_cost_per_token": float(input_cost_per_token),
            "output_cost_per_token": float(output_cost_per_token),
            "litellm_provider": "openrouter",
            "mode": "chat",
        }

        if model["per_request_limits"] is not None:
            if "prompt_tokens" in model["per_request_limits"]:
                model_data["max_input_tokens"] = int(model["per_request_limits"]["prompt_tokens"])
            if "completion_tokens" in model["per_request_limits"]:
                model_data["max_output_tokens"] = int(model["per_request_limits"]["completion_tokens"])

        if input_cost_per_image is not None and input_cost_per_image != 0.0:
            model_data["input_cost_per_image"] = float(input_cost_per_image)

        if model["architecture"]["modality"] == "multimodal":
            model_data["supports_vision"] = True

        if model_name in existing_models:
            # Update existing model with the "openrouter/" prefix
            local_data[model_name].update(model_data)
        else:
            # Add new model with the "openrouter/" prefix
            local_data[model_name] = model_data

Reply:
Adding new models is a good idea. I will see if I can fix this :) Grüße aus Australien! (Ich mache gerade ein Auslandssemester und komme aus Stuttgart hahah)